dvol_python/dvol.py

```diff
@@ -88,11 +101,13 @@ def write(self, volume, branch, commitData):
 
 
 class Voluminous(object):
-    lockFactory = DockerLock
-
-    def __init__(self, directory):
+    def __init__(self, directory, disable_docker_integration=False):
         self._directory = FilePath(directory)
         self._output = []
+        self.lockFactory = DockerLock
+        if disable_docker_integration:
+            # Do not attempt to connect to Docker if we've been asked not to.
+            self.lockFactory = NullLock
```

Comment:
Just pass in the lock factory. Booleans don't mean anything by themselves, whereas the lock factory does.